File: twao/villages/AllyVillage.java
package twao.villages;

import twao.Player;

public class AllyVillage extends Village {
    private final Player owner;

    public AllyVillage(int x, int y, Player owner) {
        super(x, y);
        this.owner = owner;
    }

    /**
     * -----------------------------------------------------------
     * getters and setters
     * -----------------------------------------------------------
     */

    public Player getOwner() { return owner; }
}

File: src/main/java/io/github/stiv3ns/twactionorganizer/twao/villages/AllyVillage.kt
package io.github.stiv3ns.twactionorganizer.twao.villages

import io.github.stiv3ns.twactionorganizer.twao.Player
import kotlinx.serialization.Serializable

data class AllyVillage(
        override val x: Int,
        override val y: Int,
        val owner: Player
) : Village {
    override var id: Int? = null
    override fun toString() = "$x|$y"
}